client/signup/site-mockup/index.jsx

```diff
@@ -66,19 +66,20 @@ class SiteMockups extends Component {
 		verticalPreviewContent: '',
 	};
 
-	shouldComponentUpdate( nextProps ) {
-		// Debouncing updates to the preview content
-		// prevents the flashing effect.
-		if ( nextProps.verticalPreviewContent !== this.props.verticalPreviewContent ) {
-			this.updateDebounced();
-			return false;
+	componentDidUpdate( prevProps ) {
+		const { siteStyle, siteType, verticalPreviewContent, verticalSlug } = this.props;
+
+		if ( ! verticalPreviewContent || prevProps.verticalPreviewContent === verticalPreviewContent ) {
+			return;
 		}
 
-		return true;
+		this.props.recordTracksEvent( 'calypso_signup_site_preview_mockup_rendered', {
+			site_type: siteType,
+			vertical_slug: verticalSlug,
+			site_style: siteStyle || 'default',
+		} );
```

Comment:
Does it make sense to amend the conditional expression to check for `true`?

```
		if (  verticalPreviewContent && prevProps.verticalPreviewContent !== verticalPreviewContent ) {
			this.props.recordTracksEvent( 'calypso_signup_site_preview_mockup_rendered', {
				site_type: siteType,
				vertical_slug: verticalSlug,
				site_style: siteStyle || 'default',
			} );
		}
```
I'm only thinking that it better describes the condition under which we should execute the internal code. 

It also means that people who wish to add to `componentDidUpdate` in the future won't need to worry about the early `return`.

Reply:
By default, I choose early returns over adding a nesting level, but agree it makes it more clear in this case to check the other direction.  Will flip.